src/channel_manager.ts

```diff
@@ -0,0 +1,517 @@
+import type { StreamChat } from './client';
+import type {
+  DefaultGenerics,
+  ExtendableGenerics,
+  Event,
+  ChannelOptions,
+  ChannelStateOptions,
+  ChannelFilters,
+  ChannelSort,
+} from './types';
+import { StateStore } from './store';
+import { Channel } from './channel';
+import {
+  findLastPinnedChannelIndex,
+  findPinnedAtSortOrder,
+  getAndWatchChannel,
+  isChannelArchived,
+  isChannelPinned,
+  moveChannelUpwards,
+  shouldConsiderArchivedChannels,
+  shouldConsiderPinnedChannels,
+} from './utils';
+
+export type ChannelManagerPagination<SCG extends ExtendableGenerics = DefaultGenerics> = {
+  filters: ChannelFilters<SCG>;
+  hasNext: boolean;
+  isLoading: boolean;
+  isLoadingNext: boolean;
+  options: ChannelOptions;
+  sort: ChannelSort<SCG>;
+  stateOptions: ChannelStateOptions;
+};
+
+export type ChannelManagerState<SCG extends ExtendableGenerics = DefaultGenerics> = {
+  channels: Channel<SCG>[];
+  pagination: ChannelManagerPagination<SCG>;
+  ready: boolean;
```

Comment:
When is `ready` flipped to `true`?

Reply:
It's done [here](https://github.com/GetStream/stream-chat-js/blob/93dec4b37d6e1e49b17e853378c6476ec5505906/src/channel_manager.ts#L223)

Reply:
@isekovanic, do we have a better name for the `ready` var? This doesn't seem to define its actual usage and is confusing.

Reply:
Yep ! Exactly. Having `channels` be `undefined` essentially means that wherever we actually use `channels` in the future we need to take this into consideration; this way, we can use `channels` directly everywhere and use `ready` (or whatever we decide to name it) wherever applicable (for example, loading state on the UI).